provider/docker.go

```diff
@@ -719,3 +765,34 @@ func parseTasks(task swarmtypes.Task, serviceDockerData dockerData, networkMap m
 	}
 	return dockerData
 }
+
+func containerExists(ctx context.Context, cli client.ContainerAPIClient, id string) bool {
+	if _, err := cli.ContainerInspect(ctx, id); err != nil {
+		return false
+	}
+	return true
+}
+
+func getNetworks(ctx context.Context, cli client.APIClient, containerID string,
```

Comment:
maybe something not that concrete, but we're having 2 ways to get the networks now (depending on the situation). I might need to think about that